cli/azd/pkg/pipeline/github_provider.go

```diff
@@ -627,18 +619,85 @@ func (p *GitHubCiProvider) configurePipeline(
 	provisioningProvider provisioning.Options,
 	additionalSecrets map[string]string,
 	additionalVariables map[string]string,
+	options *ConfigOptions,
 ) (CiPipeline, error) {
-
 	repoSlug := repoDetails.owner + "/" + repoDetails.repoName
+
+	// Variables and Secrets for a gh-actions are independent from the gh-action. They are set on the repository level.
+	// We need to clean up the previous values before setting the new ones.
+	// By doing this, we are handling:
+	// - When a secret is moved to be a variable (or vice versa). Don't leak the previous value on the pipeline.
+	// - When there was a previous additional variable/secret set and then it was updated to empty string or unset from .env.
```

Comment:
These scenarios feel like they should be manually cleaned up.  We don't know the origin of the variables/secrets were actually set by azd or not.

Reply:
we do.
Customers must explicitly define the list of secrets/variables in azure.yaml. 
azd is ONLY cleaning the the secets/variables found in the repository which matches to the list defined in the azure.yaml.